iree/tools/init_targets.h

```diff
@@ -15,32 +15,14 @@
 #ifndef IREE_TOOLS_INIT_TARGETS_H_
 #define IREE_TOOLS_INIT_TARGETS_H_
 
-#include "iree/compiler/Dialect/HAL/Target/LLVM/LLVMAOTTarget.h"
-#include "iree/compiler/Dialect/HAL/Target/LLVM/LLVMIRTarget.h"
-#include "iree/compiler/Dialect/HAL/Target/VMLA/VMLATarget.h"
-#include "iree/compiler/Dialect/HAL/Target/VulkanSPIRV/VulkanSPIRVTarget.h"
-
 namespace mlir {
 namespace iree_compiler {
 
 // This function should be called before creating any MLIRContext if one
 // expects all the possible target backends to be available. Custom tools can
 // select which targets they want to support by only registering those they
 // need.
-inline void registerHALTargetBackends() {
-  static bool init_once = []() {
-    IREE::HAL::registerLLVMAOTTargetBackends(
-        []() { return IREE::HAL::getLLVMTargetOptionsFromFlags(); });
-    IREE::HAL::registerLLVMIRTargetBackends(
-        []() { return IREE::HAL::getLLVMTargetOptionsFromFlags(); });
-    IREE::HAL::registerVMLATargetBackends(
-        []() { return IREE::HAL::getVMLATargetOptionsFromFlags(); });
-    IREE::HAL::registerVulkanSPIRVTargetBackends(
-        []() { return IREE::HAL::getVulkanSPIRVTargetOptionsFromFlags(); });
-    return true;
-  }();
-  (void)init_once;
-}
```

Comment:
Out of curiosity, why was the registration moved to `init_targets.cc` instead of simply adding the preprocessor logic to the header?

Reply:
I was seeing a linker error while working on this, which seems to have gone away now (and is lost in my local history). I'll revert that part of the change.

Reply:
https://source.cloud.google.com/results/invocations/c757d843-4653-4a94-b4dd-4ee9110e280b/targets/iree%2Fgcp_ubuntu%2Fbazel%2Flinux%2Fx86%2Fcore%2Fpresubmit/log ah, there they are... local builds weren't showing that for me.

Reply:
Ah right, https://docs.bazel.build/versions/master/be/c-cpp.html#cc_library.copts. The Bazel `copts` wouldn't affect compilation with just a header when pulled into a different library/binary, I think moving the usage into the `srcs` made the `copts` function as expected.